tests/integ/test_session.py

```diff
@@ -284,9 +284,17 @@ def test_create_session_in_sp(session):
     original_platform = internal_utils.PLATFORM
     internal_utils.PLATFORM = "XP"
     try:
-        with pytest.raises(SnowparkSessionException) as exec_info:
-            Session(session._conn)
-        assert exec_info.value.error_code == "1410"
+        if not session._conn._get_client_side_session_parameter(
+            "ENABLE_CREATE_SESSION_IN_STORED_PROCS", False
+        ):
+            with pytest.raises(SnowparkSessionException) as exec_info:
+                Session(session._conn)
+            assert exec_info.value.error_code == "1410"
+        else:
+            try:
+                Session(session._conn)
+            except SnowparkSessionException as e:
+                pytest.fail(f"Unexpected exception {e} was raised")
```

Comment:
you might also want a positive case to ensure when the session parameter is set to True, no error is raised

Reply:
Ummm I may be missing something, but isn't that what my change do? basically the else part of `if not session._conn._get_client_side_session_parameter(
            "ENABLE_CREATE_SESSION_IN_STORED_PROCS", False
        ):`

Reply:
yes, a test case that shows with `ENABLE_CREATE_SESSION_IN_STORED_PROCS` being `true` we can create the Session(session._conn).